pymbolic/interop/symengine.py

```diff
@@ -67,6 +67,16 @@ def map_ComplexDouble(self, expr):  # noqa
 
     map_RealDouble = SympyLikeToPymbolicMapper.to_float  # noqa: N815
 
+    def map_Piecewise(self, expr):  # noqa
+        # We only handle piecewises with 2 statements!
+        assert len(expr.args) == 4
+        # We only handle if/else cases
+        assert expr.args[3].is_Boolean and bool(expr.args[3]) is True
```

Comment:
It seems like these shouldn't be asserts, but `raise NotImplementedError`. (also below)